src/main/haskell/kore/src/Kore/Variables/Fresh.hs

```diff
@@ -59,19 +59,40 @@ a non-id symbol @_@ to avoid clashing with user-defined ids.
 freshVariablePrefix :: Text
 freshVariablePrefix = "var_"
 
+variableSeparator :: Text
+variableSeparator = "_"
+
 instance FreshVariable Variable where
```

Comment:
Would you mind documenting the syntax of fresh variables?  It seems to get more complex now.